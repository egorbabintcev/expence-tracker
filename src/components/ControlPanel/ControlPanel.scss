@import 'utils/functions';
@import 'utils/mixins';
@import 'utils/placeholders';

.control-panel {
  background-color: var(--color-panel-default);
  bottom: 0;
  left: 0;
  padding: rem(8);
  position: fixed;
  transition: transform 0.2s;
  transition-delay: 0.2s;
  width: 100%;

  &:not(.is-active) {
    transform: translateY(100%);
  }

  &.is-active {
    transform: translateY(0);
  }

  &__list {
    align-items: center;
    display: flex;
    justify-content: center;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__item {
    &:not(:last-child) {
      margin-right: rem(20);
    }

    button {
      align-items: center;
      background: none;
      border: 0;
      display: flex;
      flex-direction: column;
      outline: none;
      padding: 0;
    }

    &-icon {
      fill: #c4c4c4;
      height: rem(25);
      margin-bottom: rem(7);
      width: rem(25);
    }

    &-text {
      color: var(--color-text-lighter);
      line-height: rem(14);
      font: {
        size: rem(14);
        weight: 500;
      }
    }
  }
}
