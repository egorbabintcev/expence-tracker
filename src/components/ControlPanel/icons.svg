<svg xmlns="http://www.w3.org/2000/svg">
  <symbol id="create" viewBox="0 0 19 25" preserveApectRatio="xMidYMid meet">
    <path d="M18.6198 5.1269L13.7846 0.341739C13.6742 0.233009 13.5432 0.146856 13.3992 0.088207C13.2551 0.0295582 13.1008 -0.000435081 12.9449 -5.73543e-05H12.6437V6.24994H18.9655V5.95209C18.9654 5.64261 18.8411 5.34581 18.6198 5.1269V5.1269ZM11.0632 6.64057V-5.73543e-05H1.18535C1.02958 -0.000443808 0.875263 0.0296039 0.731274 0.0883592C0.587285 0.147115 0.456459 0.233419 0.346311 0.342315C0.236164 0.451211 0.148867 0.580551 0.0894365 0.722904C0.0300058 0.865257 -0.000387173 1.01782 3.72316e-06 1.17182V23.8281C-0.000387173 23.9821 0.0300058 24.1346 0.0894365 24.277C0.148867 24.4193 0.236164 24.5487 0.346311 24.6576C0.456459 24.7665 0.587285 24.8528 0.731274 24.9115C0.875263 24.9703 1.02958 25.0003 1.18535 24.9999H17.7802C17.9359 25.0003 18.0902 24.9703 18.2342 24.9115C18.3782 24.8528 18.5091 24.7665 18.6192 24.6576C18.7293 24.5487 18.8166 24.4193 18.8761 24.277C18.9355 24.1346 18.9659 23.9821 18.9655 23.8281V7.81244H12.2486C11.9345 7.81154 11.6335 7.68779 11.4114 7.46821C11.1893 7.24864 11.0641 6.95109 11.0632 6.64057V6.64057ZM14.6192 15.2343V16.0156C14.6192 16.2228 14.536 16.4215 14.3878 16.568C14.2396 16.7145 14.0386 16.7968 13.829 16.7968H10.6681V19.9218C10.6681 20.129 10.5848 20.3277 10.4366 20.4742C10.2884 20.6208 10.0875 20.7031 9.87787 20.7031H9.08764C8.87806 20.7031 8.67706 20.6208 8.52886 20.4742C8.38067 20.3277 8.29741 20.129 8.29741 19.9218V16.7968H5.13649C4.92691 16.7968 4.72591 16.7145 4.57772 16.568C4.42952 16.4215 4.34627 16.2228 4.34627 16.0156V15.2343C4.34627 15.0271 4.42952 14.8284 4.57772 14.6819C4.72591 14.5354 4.92691 14.4531 5.13649 14.4531H8.29741V11.3281C8.29741 11.1209 8.38067 10.9222 8.52886 10.7756C8.67706 10.6291 8.87806 10.5468 9.08764 10.5468H9.87787C10.0875 10.5468 10.2884 10.6291 10.4366 10.7756C10.5848 10.9222 10.6681 11.1209 10.6681 11.3281V14.4531H13.829C14.0386 14.4531 14.2396 14.5354 14.3878 14.6819C14.536 14.8284 14.6192 15.0271 14.6192 15.2343Z" />
  </symbol>
  <symbol id="delete" viewBox="0 0 22 25" preserveApectRatio="xMidYMid meet">
    <path d="M1.55279 22.6563C1.55279 23.2779 1.79819 23.874 2.235 24.3135C2.67181 24.7531 3.26425 25 3.88199 25H17.8571C18.4749 25 19.0673 24.7531 19.5041 24.3135C19.9409 23.874 20.1863 23.2779 20.1863 22.6563V6.25001H1.55279V22.6563ZM14.7515 10.1563C14.7515 9.94905 14.8333 9.75034 14.9789 9.60383C15.1246 9.45731 15.322 9.375 15.5279 9.375C15.7339 9.375 15.9313 9.45731 16.0769 9.60383C16.2225 9.75034 16.3043 9.94905 16.3043 10.1563V21.0938C16.3043 21.301 16.2225 21.4997 16.0769 21.6462C15.9313 21.7927 15.7339 21.875 15.5279 21.875C15.322 21.875 15.1246 21.7927 14.9789 21.6462C14.8333 21.4997 14.7515 21.301 14.7515 21.0938V10.1563ZM10.0932 10.1563C10.0932 9.94905 10.175 9.75034 10.3206 9.60383C10.4662 9.45731 10.6636 9.375 10.8696 9.375C11.0755 9.375 11.273 9.45731 11.4186 9.60383C11.5642 9.75034 11.646 9.94905 11.646 10.1563V21.0938C11.646 21.301 11.5642 21.4997 11.4186 21.6462C11.273 21.7927 11.0755 21.875 10.8696 21.875C10.6636 21.875 10.4662 21.7927 10.3206 21.6462C10.175 21.4997 10.0932 21.301 10.0932 21.0938V10.1563ZM5.43478 10.1563C5.43478 9.94905 5.51658 9.75034 5.66218 9.60383C5.80778 9.45731 6.00526 9.375 6.21118 9.375C6.41709 9.375 6.61457 9.45731 6.76017 9.60383C6.90578 9.75034 6.98758 9.94905 6.98758 10.1563V21.0938C6.98758 21.301 6.90578 21.4997 6.76017 21.6462C6.61457 21.7927 6.41709 21.875 6.21118 21.875C6.00526 21.875 5.80778 21.7927 5.66218 21.6462C5.51658 21.4997 5.43478 21.301 5.43478 21.0938V10.1563ZM20.9627 1.56251H15.1397L14.6836 0.649422C14.587 0.454214 14.4381 0.29001 14.2538 0.175281C14.0695 0.0605526 13.8571 -0.00014785 13.6403 8.56089e-06H8.09394C7.8777 -0.000827891 7.66561 0.0596462 7.48195 0.174502C7.29829 0.289359 7.1505 0.453951 7.05551 0.649422L6.59938 1.56251H0.776397C0.570484 1.56251 0.373004 1.64482 0.227401 1.79133C0.0817988 1.93784 0 2.13656 0 2.34376L0 3.90626C0 4.11346 0.0817988 4.31217 0.227401 4.45868C0.373004 4.6052 0.570484 4.68751 0.776397 4.68751H20.9627C21.1686 4.68751 21.3661 4.6052 21.5117 4.45868C21.6573 4.31217 21.7391 4.11346 21.7391 3.90626V2.34376C21.7391 2.13656 21.6573 1.93784 21.5117 1.79133C21.3661 1.64482 21.1686 1.56251 20.9627 1.56251V1.56251Z" />
  </symbol>
  <symbol id="settings" viewBox="0 0 28 25" preserveApectRatio="xMidYMid meet">
    <path d="M26.3247 19.2935H8.49185V18.4443C8.49185 17.9772 8.10971 17.5951 7.64266 17.5951H5.94429C5.47724 17.5951 5.09511 17.9772 5.09511 18.4443V19.2935H0.849185C0.382133 19.2935 0 19.6756 0 20.1427V21.841C0 22.3081 0.382133 22.6902 0.849185 22.6902H5.09511V23.5394C5.09511 24.0065 5.47724 24.3886 5.94429 24.3886H7.64266C8.10971 24.3886 8.49185 24.0065 8.49185 23.5394V22.6902H26.3247C26.7918 22.6902 27.1739 22.3081 27.1739 21.841V20.1427C27.1739 19.6756 26.7918 19.2935 26.3247 19.2935ZM26.3247 10.8016H22.0788V9.95245C22.0788 9.4854 21.6967 9.10327 21.2296 9.10327H19.5313C19.0642 9.10327 18.6821 9.4854 18.6821 9.95245V10.8016H0.849185C0.382133 10.8016 0 11.1838 0 11.6508V13.3492C0 13.8162 0.382133 14.1984 0.849185 14.1984H18.6821V15.0476C18.6821 15.5146 19.0642 15.8967 19.5313 15.8967H21.2296C21.6967 15.8967 22.0788 15.5146 22.0788 15.0476V14.1984H26.3247C26.7918 14.1984 27.1739 13.8162 27.1739 13.3492V11.6508C27.1739 11.1838 26.7918 10.8016 26.3247 10.8016ZM26.3247 2.30979H15.2853V1.4606C15.2853 0.993553 14.9032 0.61142 14.4361 0.61142H12.7378C12.2707 0.61142 11.8886 0.993553 11.8886 1.4606V2.30979H0.849185C0.382133 2.30979 0 2.69192 0 3.15897V4.85734C0 5.3244 0.382133 5.70653 0.849185 5.70653H11.8886V6.55571C11.8886 7.02277 12.2707 7.4049 12.7378 7.4049H14.4361C14.9032 7.4049 15.2853 7.02277 15.2853 6.55571V5.70653H26.3247C26.7918 5.70653 27.1739 5.3244 27.1739 4.85734V3.15897C27.1739 2.69192 26.7918 2.30979 26.3247 2.30979Z" />
  </symbol>

  <symbol id="select-all" viewBox="0 0 384 512" preserveAspectRatio="xMidYMid meet">
    <path d="M360 0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V24c0-13.3-10.7-24-24-24zM64 112c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-32zm1.6 129.4l12.7-12.6c2.1-2.1 5.5-2.1 7.6 0l20.6 20.8 47.6-47.2c2.1-2.1 5.5-2.1 7.6 0l12.6 12.7c2.1 2.1 2.1 5.5 0 7.6l-64.2 63.6c-2.1 2.1-5.5 2.1-7.6 0L65.6 249c-2.1-2.1-2.1-5.5 0-7.6zM128 400c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-32c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32zm192-8c0 4.4-3.6 8-8 8H168c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16zm0-128c0 4.4-4.3 8-9.6 8H170.2s29.2-30.2 30.4-32h109.7c5.3 0 9.6 3.6 9.6 8v16zm0-128c0 4.4-3.6 8-8 8H168c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16z" />
  </symbol>

  <symbol id="cancel" viewBox="0 0 320 512" preserveAspectRatio="xMidYMid slice">
    <path d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z" />
  </symbol>
</svg>
