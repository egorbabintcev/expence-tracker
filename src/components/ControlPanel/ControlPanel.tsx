import React, { useEffect, useState } from 'react'
import cx from 'classnames'
import { modalStore } from 'core/stores'
import './ControlPanel.scss'
import icons from './icons.svg'

/*
interface ControlPanelProps {

}
*/

const ControlPanel: React.FC = () => {
  const [isActive, setIsActive] = useState(true)
  const { openModal } = modalStore

  let prevScroll = window.scrollY
  const handleScroll: EventListener = (e: Event) => {
    const target = e.currentTarget as Window | HTMLElement

    let property: string
    if (target instanceof Window) {
      property = 'scrollY'
    } else {
      property = 'scrollTop'
    }

    if (prevScroll > target[property]) {
      setIsActive(true)
    } else if (prevScroll < target[property]) {
      setIsActive(false)
    }
    prevScroll = target[property]
  }

  useEffect(() => {
    window.addEventListener('scroll', handleScroll)

    return () => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  return (
    <div className={cx('control-panel', { 'is-active': isActive })}>
      <ul className="control-panel__list">
        <li className="control-panel__item">
          <button type="button" onClick={openModal.bind(null, 'editor')}>
            <svg className="control-panel__item-icon">
              <use xlinkHref={`${icons}#create`} />
            </svg>
            <span className="control-panel__item-text">Добавить</span>
          </button>
        </li>
        <li className="control-panel__item">
          <button type="button">
            <svg className="control-panel__item-icon">
              <use xlinkHref={`${icons}#delete`} />
            </svg>
            <span className="control-panel__item-text">Удалить</span>
          </button>
        </li>
        <li className="control-panel__item">
          <button type="button">
            <svg className="control-panel__item-icon">
              <use xlinkHref={`${icons}#settings`} />
            </svg>
            <span className="control-panel__item-text">Настройки</span>
          </button>
        </li>
      </ul>
    </div>
  )
}

export default ControlPanel
