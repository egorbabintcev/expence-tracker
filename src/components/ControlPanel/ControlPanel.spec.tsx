import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import ControlPanel from './'

describe('ControlPanel component', () => {
  test('Should match snapshot', () => {
    const { container } = render(<ControlPanel />)
    expect(container).toMatchSnapshot()
  })
})
