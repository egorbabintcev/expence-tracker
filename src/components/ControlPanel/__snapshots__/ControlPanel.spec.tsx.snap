// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel component Should match snapshot 1`] = `
<div>
  <div
    class="control-panel is-active"
  >
    <ul
      class="control-panel__list"
    >
      <li
        class="control-panel__item"
      >
        <button
          type="button"
        >
          <svg
            class="control-panel__item-icon"
          >
            <use
              xlink:href="icons.svg#create"
            />
          </svg>
          <span
            class="control-panel__item-text"
          >
            Добавить
          </span>
        </button>
      </li>
      <li
        class="control-panel__item"
      >
        <button
          type="button"
        >
          <svg
            class="control-panel__item-icon"
          >
            <use
              xlink:href="icons.svg#delete"
            />
          </svg>
          <span
            class="control-panel__item-text"
          >
            Удалить
          </span>
        </button>
      </li>
      <li
        class="control-panel__item"
      >
        <button
          type="button"
        >
          <svg
            class="control-panel__item-icon"
          >
            <use
              xlink:href="icons.svg#settings"
            />
          </svg>
          <span
            class="control-panel__item-text"
          >
            Настройки
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
